### PR TITLE
update Release History

### LINKED ISSUE #220: update Release History
Update the [Release History](https://qiskit.org/documentation/release_history.html) doc page using the change logs of each Qiskit element and a synopsis.

- [Terra](https://github.com/Qiskit/qiskit-terra/blob/master/CHANGELOG.rst)
- [Aer](https://github.com/Qiskit/qiskit-aer/blob/master/CHANGELOG.rst)
- [Aqua](https://github.com/Qiskit/qiskit-aqua/blob/master/CHANGELOG.rst)
- [Ignis](https://github.com/Qiskit/qiskit-ignis/blob/master/CHANGELOG.md)

### PR DESCRIPTION
**Pull request by derivation**
closes #220

* Update release history for Aer and Aqua in docs based on their change logs
* Add `extlinks` Sphinx extension to `conf.py` to easily link to pull requests in `release_history.rst`

### CODE REVIEW

File: docs/release_history.rst [hunk 1]
```diff
@@ -111,6 +111,152 @@ adjust for before the next release to avoid a breaking change.
...
+* Added capability to create a ``Custom`` initial state using existing
+  circuit.
+* Added the ADAM (and AMSGRAD) optimization algorithm
+* Multivariate distributions added, so uncertainty models now have univariate
+  and multivariate distribution components.
+* Added option to include or skip the swaps operations for qft and iqft
+  circuit constructions.
+* Added classical linear system solver ``ExactLSsolver``.
+* Added parameters ``auto_hermitian`` and ``auto_resize`` to ``HHL`` algorithm
+  to support non-hermititan and non 2**n sized matrices by default.
```
Comment (lzdanski) [inline docs/release_history.rst]:
```suggestion
  to support non-hermititan and non 2\**n sized matrices by default.
```

Comment (lzdanski) [reply docs/release_history.rst]: There's something funky happening with the .rst formatting here. The two `**` will cause the text to bold. Does this need to be `2**n` or `2^n`? Either way, we might need the escape character. However... with markdown preview... it's making things italicize? Maybe worth checking in a local build.

File: docs/release_history.rst [hunk 2]
```diff
@@ -111,6 +111,152 @@ adjust for before the next release to avoid a breaking change.
...
 
+
+
+Aer 0.2
+~~~~~~~
+
+New Features
+^^^^^^^^^^^^
+
+* Add multiplexer gate :pull_aer:`192`
```
Comment (lzdanski) [inline docs/release_history.rst]: This is super minor... but some of these are in present tense and others are in past tense? Add vs Added. It's a nice to have for consistency.

**Review by lzdanski — DISMISSED** [review]
Looks good! Only one thing really needs any close attention. The other is a formatting/style choice.

File: docs/release_history.rst (hunk 1 above)
Comment (derivation) [reply docs/release_history.rst]: I'll use the :math: role.

File: docs/release_history.rst (hunk 2 above)
Comment (derivation) [reply docs/release_history.rst]: I'll have things in the past tense.

**Review by jaygambetta — APPROVED** [review]

**Pull request merged**